packages/caliper-core/test/worker/rate-control/fixedFeedbackRate.js

```diff
@@ -108,41 +108,176 @@ describe('fixedFeedbackRate controller implementation', () => {
+            sinon.stub(txnStats, 'getTotalFinishedTx').returns(0);
 
-            // should have not called the sleep method
+            await controller.applyRateControl();
+
+            // Ensure sleep was not called
             sinon.assert.notCalled(sleepStub);
         });
 
-        it('should not sleep if backlog transaction is below half the target', () => {
+        it('should not sleep if backlog transaction is below half the target', async () => {
             txnStats.stats.txCounters.totalSubmitted = 1000;
             txnStats.stats.txCounters.totalFinished = 999;
             controller.generalSleepTime = 1;
-            controller.applyRateControl();
+            await controller.applyRateControl();
 
             // should have not called the sleep method
             sinon.assert.notCalled(sleepStub);
         });
 
-        it ('should sleep if the elapsed time difference is greater than 5ms', () => {
+        it ('should sleep if the elapsed time difference is greater than 5ms', async () => {
             txnStats.stats.txCounters.totalSubmitted = 100;
             txnStats.stats.txCounters.totalFinished = 2;
-            controller.applyRateControl();
+            await controller.applyRateControl();
 
             // should have called the sleep method with a value equal to diff
             sinon.assert.calledOnce(sleepStub);
             sinon.assert.calledWith(sleepStub, 20000);
         });
+
+        it('should sleep for backOffTime when no successful transactions have occurred once', async () => {
+            txnStats.stats.txCounters.totalSubmitted = 5;
+            txnStats.stats.txCounters.totalFinished = 2;
+            txnStats.stats.txCounters.totalSuccessful = 0;
+            controller.generalSleepTime = 1;
+
+            await controller.applyRateControl();
+
+            sinon.assert.calledOnce(sleepStub);
+            sinon.assert.calledWith(sleepStub, 1 * controller.backOffTime);
+        });
+
+        it('should increase sleep time when no successful transactions have occurred multiple times', async () => {
+            // Configure the rate controller with a high TPS value
+            testMessage.content.rateControl.opts = {
+                tps: 10000,
+                sleepTime: 100,
+                transactionLoad: 10
+            };
+
+            controller = new FixedFeedbackRate.createRateController(testMessage, txnStats, 0);
+            txnStats.stats.txCounters.totalSubmitted = 5;
+            txnStats.stats.txCounters.totalFinished = 2;
+            txnStats.stats.txCounters.totalSuccessful = 0;
+
+            controller.zeroSuccessfulCounter = 3;
+            clock.tick(0); // Ensure clock starts at 0
+
+            await controller.applyRateControl();
+
+            sinon.assert.calledOnce(sleepStub);
+            // Expected sleep time should be 4 * backOffTime = 400ms
+            sinon.assert.calledWith(sleepStub, 4 * controller.backOffTime);
```

Comment:
This is 1 more than the zeroSuccessfulCounter, would be good to show this in this test